share/availability/discovery/options.go

```diff
@@ -0,0 +1,69 @@
+package discovery
+
+import (
+	"fmt"
+	"time"
+)
+
+// Parameters is the set of Parameters that must be configured for the Discovery module
+type Parameters struct {
+	// PeersLimit defines the soft limit of FNs to connect to via discovery.
+	// Set 0 to disable.
+	PeersLimit uint
+	// AdvertiseInterval is a interval between advertising sessions.
+	// Set -1 to disable.
+	// NOTE: only full and bridge can advertise themselves.
+	AdvertiseInterval time.Duration
+	// discoveryRetryTimeout is an interval between discovery attempts
+	// when we discovered lower than PeersLimit peers.
+	discoveryRetryTimeout time.Duration
+}
+
+// Option is a function that configures Discovery Parameters
+type Option func(*Parameters)
+
+// DefaultParameters returns the default Parameters' configuration values
+// for the Discovery module
+func DefaultParameters() Parameters {
+	return Parameters{
```

Comment:
why isn't the default for discoveryRetryTimeout included here?

This will panic the ticker.

Reply:
It will panic for sure.  discoveryRetryTimeout  should be overwritten by defaults in constructor the same way it was before. Currently we read defaults from config file, not from code and private values could not be populated this way. So the simplest way to make local params work, without major config defaults refactoring is to overwrite.

Reply:
defaultRetryTimeout